providers/check.rb

```diff
@@ -29,8 +29,15 @@ def load_current_resource
       timeout
       ttl
       type
+      interval
+      cron
     ]
-  ).merge("interval" => new_resource.interval).merge(new_resource.additional)
+  ).merge(new_resource.additional).tap do |helper|
+    if helper.key?("interval") && helper.key?("cron")
```

Comment:
you probably will need to adjust this based on the change to interval, you probably would want to remove the key from the hash if its value is `false` or `!positive?` (which would account for `0` and `negative` values)